client/pages/editTemplate.jsx

```diff
@@ -0,0 +1,247 @@
+import React, { useEffect, useState } from "react";
+import {
+  addTemplate,
+  saveTemplateName,
+  saveTemplatePhases,
+  deleteTemplate,
+  getTemplateByID,
+  getTemplates
+} from "../utils/apiWrapper";
+import { Head, TemplateDraft } from "../components";
+import {
+  Button,
+  Row,
+  Container,
+  Alert,
+  Form,
+  FormGroup,
+  Input,
+  Label
+} from "reactstrap";
+import "../public/styles/editTemplate.scss";
+
+export default function EditTemplate() {
+  const [name, setName] = useState("");
+  const [mounted, setMounted] = useState(false);
+  const [templateID, setTemplateID] = useState("");
+  const [error, setError] = useState(false);
+  const [templates, setTemplates] = useState([]);
+  const [isInspiration, setInspiration] = useState(false);
+  const [isIdeation, setIdeation] = useState(false);
+  const [isImplementation, setImplementation] = useState(false);
+
+  const phases = ["Inspiration", "Ideation", "Implementation"];
+  const implementationIndex = 2;
+
+  useEffect(() => {
+    const setDisplay = async () => {
+      const currentTemplates = await getTemplates();
+      setTemplates(currentTemplates);
+
+      if (currentTemplates.data.length === 0) {
+        setName("");
+        setInspiration(false);
+        setIdeation(false);
+        setImplementation(false);
+      } else {
+        let result;
+
+        if (!mounted) {
+          setTemplateID(currentTemplates.data[0]._id);
+          setMounted(true);
+
+          result = await getTemplateByID(currentTemplates.data[0]._id);
+        } else {
+          result = await getTemplateByID(templateID);
+        }
+
+        if (result.data !== undefined && result) {
+          if (result.data.name !== undefined) {
+            setName(result.data.name);
+          } else {
+            setName("");
+          }
+          if (result.data.phases !== undefined) {
+            setInspiration(result.data.phases.includes(phases[0]));
+            setIdeation(result.data.phases.includes(phases[1]));
+            setImplementation(
+              result.data.phases.includes(phases[implementationIndex])
+            );
+          } else {
+            setInspiration(false);
+            setIdeation(false);
+            setImplementation(false);
+          }
+        }
+      }
+    };
+
+    setDisplay();
+  }, [templateID]);
+
+  const handleID = clickedTemplateID => {
+    setTemplateID(clickedTemplateID);
+  };
+
+  const handleNewStage = async e => {
+    e.preventDefault();
```

Comment:
Why do you need to prevent the default action of the event? Might not need this.